### PR TITLE
feat: decouple keycloak database from the community Helm chart

### PR DESCRIPTION
**Pull request by j-zimnowoda**
## Checklist

- [ ] Architecture Design Records have been added as `adr/*.md` and appended to list in `adr/_index.md`, if applicable.
- [ ] The `values-schema.yaml` file and `test/**` fixtures have been updated to reflect code changes, if applicable.
- [ ] The OpenApi Schema from redkubes/otomi-api project is compatible with definitions from `values-schema.yaml` file, if applicable.
- [ ] Helm releases are meeting otomi's baseline security policies, if applicable.
- [ ] Helm chart and helmfile changes are tested against upgrade scenario, if applicable.

### CODE REVIEW

File: helmfile.d/snippets/defaults.yaml [hunk 1]
```diff
@@ -625,6 +625,12 @@ environments:
               limits:
                 cpu: 1000m
                 memory: 512Mi
+        databases:
+          keycloak:
+            useOtomiDB: true
+            imported: false
+            size: "1Gi"
+            replicas: 2
```
Comment (j-zimnowoda) [inline helmfile.d/snippets/defaults.yaml:633]: Let's make replicas 1 by default, so users that evaluate Otomi by default does not consume too many resources 
replicas: 1

File: values/keycloak/keycloak-otomi-db.gotmpl [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+      localeCType: 'en_US.UTF-8'
+{{- end }}
+
+instances: {{ $kdb.replicas }}
+
+resources:
+  {{- with $kdb | get "resources" nil }}
+  {{- toYaml . | nindent 6 }}
+  {{- else }}
+  limits:
```
Comment (j-zimnowoda) [inline values/keycloak/keycloak-otomi-db.gotmpl:49]: Could you move default resources to the `defaults.yaml` file?
Then in this file there will be a simple reference:
```
resources: {{- toYaml $kdb.resources | nindent 6 }}
```

**Review by j-zimnowoda — COMMENTED** [review]
Looking good, just few remarks and we are one step farther in decoupling databases from the community charts :)

**Comment by CasLubbers** [comment]
Agreed with Jeho, lgtm

**Review by CasLubbers — APPROVED** [review]

File: values/keycloak/keycloak.gotmpl [hunk 3]
```diff
@@ -87,6 +88,39 @@ extraVolumes: |
...
+    value: keycloak
+{{- else }}
+extraEnv: |
+  - name: KEYCLOAK_USER
+    value: {{ $k.adminUsername }}
+  - name: KEYCLOAK_PASSWORD
+    value: {{ $k.adminPassword }}
+{{- end }}
+
+{{- if $v.databases.keycloak.useOtomiDB }}
```
Comment (CasLubbers) [inline values/keycloak/keycloak.gotmpl:119]: This could be in the same if statement above, but this is better for readability I think.

**Review by Ani1357 — APPROVED** [review]

**Pull request merged**